File: code-samples/team-messaging/code-snippets/add-team-members.cs
using System;
using System.IO;
using System.Threading.Tasks;
using System.Collections.Generic;
using RingCentral;
using Newtonsoft.Json;

namespace AddTeamMembers {
  class Program {
    static RestClient restClient;

    static async Task Main(string[] args){
      // Instantiate the SDK
      restClient = new RestClient("SANDBOX-APP-CLIENT-ID", "SANDBOX-APP-CLIENT-SECRET", "https://platform.devtest.ringcentral.com");
      // Authenticate a user using a personal JWT token
      await restClient.Authorize("SANDBOX-JWT");

      await find_team("", "C# Team");
    }
    /*
    * Find the team id of the team to be added new members
    */
    static private async Task find_team(String pageToken, String teamName)
    {
      try
      {
        var queryParams = new ListGlipTeamsNewParameters()
        {
          recordCount = 10,
          pageToken = pageToken
        };

        var resp = await restClient.TeamMessaging().V1().Teams().List(queryParams);
        // Search through the team list to find the team
        Console.WriteLine("Find the team id of the " + teamName);
        foreach (var record in resp.records)
        {
          if (record.name == teamName)
          {
            Console.WriteLine("Add new members to team \"" + teamName + "\"");
            await add_new_members(record.id);
            return;
          }
        }
        // To read the next page, check and use the previous page token in the navigation object.
        if (resp.navigation.prevPageToken != null)
        {
          Thread.Sleep(1200);
          Console.WriteLine("Read newer teams ...");
          await find_team(resp.navigation.prevPageToken, teamName);
        }
        else
        {
          Console.WriteLine("Cannot find team " + teamName);
        }
      }
      catch (Exception ex)
      {
        Console.WriteLine("Unable to read teams. " + ex.Message);
      }
    }
    /*
    * Add new members to a team identified by the team id
    */
    static private async Task add_new_members(String teamId)
    {
      try
      {
        var bodyParams = new TMAddTeamMembersRequest();

        // Add internal members using their extension id
        bodyParams.members = new TMAddTeamMembersRequestMembers[] {
          // replace the email addresses below with valid internal or external new member email address
          new TMAddTeamMembersRequestMembers { email = "member.name@abc.com" },
          new TMAddTeamMembersRequestMembers { email = "guest.name@xyz.com" },
        };

        var resp = await restClient.TeamMessaging().V1().Teams(teamId).Add().Post(bodyParams);
        Console.WriteLine("New member(s) added.");
      }
      catch (Exception ex)
      {
        Console.WriteLine("Unable to create a new team. " + ex.Message);
      }
    }
  }
}

File: code-samples/team-messaging/code-snippets/add-team-members.java
package AddTeamMembers;

import java.io.IOException;
import com.google.common.reflect.TypeToken;
import com.google.gson.Gson;

import com.ringcentral.*;
import com.ringcentral.definitions.*;

public class AddTeamMembers {
    static RestClient restClient;

    public static void main(String[] args) {
      var obj = new AddTeamMembers();
      try {
        // Instantiate the SDK
        restClient = new RestClient("RC_APP_CLIENT_ID", "RC_APP_CLIENT_SECRET", "https://platform.ringcentral.com");

        // Authenticate a user using a personal JWT token
        restClient.authorize("RC_USER_JWT");
        obj.find_team("", "Java Team");
      } catch (RestException e) {
        System.out.println(e.getMessage());
      } catch (IOException e) {
        e.printStackTrace();
      }
    }
    /*
    * Find the team id of the team to be added new members
    */
  	public void find_team(String pageToken, String teamName) throws RestException, IOException {
  		try {
  			var queryParams = new ListGlipTeamsNewParameters()
  					.recordCount(10l)
  					.pageToken(pageToken);

  			var resp = restClient.teamMessaging().v1().teams().list(queryParams);
  			// Search through the team list to find the team
  	    System.out.println("Find the team id of the " + teamName);
  			for (var record : resp.records) {
  				if (record.name.equals(teamName)){
  		            System.out.println("Add new members to team \"" + teamName + "\"");
  		            add_new_members(record.id);
  		            return;
  		        }
  			}
  			// To read the next page, check and use the previous page token in the navigation object.
  			if (resp.navigation.prevPageToken != null) {
  				try {
  					Thread.sleep(1200);
  					System.out.println("Read newer teams ...");
  					find_team(resp.navigation.prevPageToken, teamName);
  				} catch (InterruptedException e) {
  					e.printStackTrace();
  				}
  			}else {
  				System.out.println("Cannot find team " + teamName);
  			}
  		}catch (RestException e){
  			System.out.println(e.getMessage());
  		}
  	}
  	/*
    * Add new members to a team identified by the team id
    */
  	public void add_new_members(String teamId) throws RestException, IOException{
  		try {
  			var bodyParams = new TMAddTeamMembersRequest();
  	        // Add internal members using their extension id
  	    bodyParams.members = new TMAddTeamMembersRequestMembers[] {
  	          // replace the email addresses below with valid internal or external new member email address
  	          new TMAddTeamMembersRequestMembers().email("member.name@abc.com"),
  	          new TMAddTeamMembersRequestMembers().email("guest.name@xyz.com")
  	    };

  	    restClient.teamMessaging().v1().teams(teamId).add().post(bodyParams);
  	    System.out.println("New member(s) added.");
  		} catch (RestException e) {
  			System.out.println(e.getMessage());
  		}

  	}
}
